src/hooks/useChain.ts

```diff
@@ -0,0 +1,20 @@
+import { useLocation } from 'react-router-dom';
+import { useGetNetworkConfig } from '@multiversx/sdk-dapp/hooks';
+import { NETWORK } from '../constants';
+import { EnvironmentsEnum } from '@multiversx/sdk-dapp/types';
+import { useMemo } from 'react';
+import { getEnvironmentForChainId } from '@multiversx/sdk-dapp/apiCalls/configuration/getEnvironmentForChainId';
+
+export const useChain = () => {
+  const { search } = useLocation();
+  const { network } = useGetNetworkConfig();
+
+  const entries = Object.fromEntries(new URLSearchParams(search));
+  const environment = entries[NETWORK] as EnvironmentsEnum;
+  const chain = useMemo<EnvironmentsEnum>(
+    () => environment || getEnvironmentForChainId(network.chainId),
+    [environment, network.chainId]
+  );
+
+  return { chain };
```

Comment:
Why return an object if it has only one key?

Reply:
We could turn it into an object when the need arrives, no needed to future proof something that might not happen. 👍